### DISCUSSION TITLE
Profiling / tracing user mode shared libraries

### DISCUSSION THREAD

**Discussion by rerrabolu** [bpftrace/bpftrace#1761] [Ideas]
I am wondering if you could suggest a simple article that shows how one can use BPFTRACE to collect profiling/tracing data of a user mode library. The approach should be applicable to any user mode shared library and not restricted to those that are widely used and are considered system libraries - e.g. C library, etc.

I have spent nearly 6 plus hours and have not come across a simple tutorial / page. It seems the articles I came across assume users already know the terminology being used and that they can distinguish tracepoints, probes, etc

**Comment by fbs** [comment]
If I read this right you're mostly confused about the tracing "fundamentals" right? So what probe types do/mean and how they hook into the system?

Have you tried searching for linux tracing in general? You might be able to find a bit more about bcc tracing.

**Comment by danobi** [comment]
Not sure if you read these yet:

* https://github.com/iovisor/bpftrace/blob/master/docs/reference_guide.md#3-uprobeuretprobe-dynamic-tracing-user-level
* https://github.com/iovisor/bpftrace/blob/master/docs/reference_guide.md#4-uprobeuretprobe-dynamic-tracing-user-level-arguments

But the same principles can be applied to any library. Not sure why 

> to any user mode shared library and not restricted to those that are widely used and are considered system libraries - e.g. C library, etc.

is applicable here
